### PR TITLE
Unify config parsers

### PR DESCRIPTION
**Pull request by saswatamcode**
This PR unifies the config parser of `transform --config` and `fmt --links.validate.config` flag into one package and changes `transform --config` flag to `pathorcontent`. 

Currently, two separate files need to be provided each for transform and link validation (even if two YAML docs are specified in one file, it will fail based on the order of docs). With this change, it will allow user to specify both transform and link validation config in one YAML file.

### CODE REVIEW

File: pkg/config/config.go [hunk 1]
```diff
@@ -48,10 +57,13 @@ type Config struct {
...
 	// LocalLinksStyle sets linking style to be applied. If empty, we assume default style.
 	LocalLinksStyle LocalLinksStyle `yaml:"localLinksStyle"`
+
+	// Validators set multiple composable validators for link checking.
+	Validators []ValidatorConfig `yaml:"validators"`
 }
 
 type TransformationConfig struct {
-	_glob glob.Glob
+	ParsedGlob glob.Glob
```
Comment (bwplotka) [inline pkg/config/config.go:66]:
```suggestion
	parsedGlob glob.Glob
```
If you make it public it will be parsable by yaml decoders (and other decoders )

**Review by bwplotka — CHANGES_REQUESTED** [review]
Hmmmmmmm I am not sure of this idea.

I would keep transform and validators in separate package configs really. Those are totally orthogonal things to configure, no?

File: pkg/config/config.go (hunk 1 above)
Comment (saswatamcode) [reply pkg/config/config.go:66]: Sure! Also, these are both totally orthogonal things but as it is currently, a user would have to specify two different YAML files(if they choose not to go with just passing content) to make both transform and format work. Would that be okay? Then I can close this PR. 🙂

Comment (bwplotka) [reply pkg/config/config.go:66]: Yes, why not? Those are two separate logics (:

Comment (bwplotka) [reply pkg/config/config.go:66]: Two different commands even.

Comment (saswatamcode) [reply pkg/config/config.go:66]: Okay then! 🙂

**Pull request closed without merging**